sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ColumnDefinition.scala

```diff
@@ -203,7 +206,9 @@ case class DefaultValueExpression(child: Expression, originalSQL: String)
   // Convert the default expression to ColumnDefaultValue, which is required by DS v2 APIs.
   def toV2(statement: String, colName: String): ColumnDefaultValue = child match {
     case Literal(value, dataType) =>
-      new ColumnDefaultValue(originalSQL, LiteralValue(value, dataType))
+      val currentDefault = analyzedChild.flatMap(new V2ExpressionBuilder(_).build())
```

Comment:
@viirya, here is the difference between current and exists default. The current default is not necessarily a literal.